### ISSUE TITLE
resnet_ pyconv and resnet_ pyconv_ at

### ISSUE THREAD

**Issue by yan-ctrl** [QingbeiGuo/SDA-xNet#1]
Hello, thank you for your excellent work. Is your SDA-xNet in this script right? https://github.com/QingbeiGuo/SDA-xNet/blob/master/SDA-PyConv/models/resnet_pyconv/resnet_pyconv.py

I want to ask you: resnet_ pyconv and resnet_ pyconv_ at, what is the difference between the two scripts

**Comment by QingbeiGuo** [comment]
resnet_ pyconv code is from https://github.com/iduta/pyconv, and the other resnet_ pyconv_at is our code. The difference between the two scripts is 

class BasicStage(nn.Module):
    def __init__(self, block, inplanes, planes, blocks, stride=1, norm_layer=None, pyconv_kernels=[3], pyconv_groups=[1], r=16, L=64):
        super(BasicStage, self).__init__()

        if norm_layer is None:
            norm_layer = nn.BatchNorm2d
        downsample = None
        if stride != 1 and inplanes != planes * block.expansion:
            downsample = nn.Sequential(
                nn.MaxPool2d(kernel_size=3, stride=stride, padding=1),
                conv1x1(inplanes, planes * block.expansion),
                norm_layer(planes * block.expansion),
            )
        elif inplanes != planes * block.expansion:
            downsample = nn.Sequential(
                conv1x1(inplanes, planes * block.expansion),
                norm_layer(planes * block.expansion),
            )
        elif stride != 1:
            downsample = nn.MaxPool2d(kernel_size=3, stride=stride, padding=1)

        layers = []
        layers.append(block(inplanes, planes, stride=stride, downsample=downsample, norm_layer=norm_layer,
                            pyconv_kernels=pyconv_kernels, pyconv_groups=pyconv_groups))
        inplanes = planes * block.expansion
        for _ in range(1, blocks):
            layers.append(block(inplanes, planes, norm_layer=norm_layer,
                                pyconv_kernels=pyconv_kernels, pyconv_groups=pyconv_groups))
        self.layers =  nn.Sequential(*layers)

        **_d = max(int(planes * block.expansion/r), L)
        self.gap = nn.AdaptiveAvgPool2d((1,1))
        self.fc = nn.Sequential(nn.Conv2d(planes * block.expansion, d, kernel_size=1, stride=1, bias=False),
                                nn.BatchNorm2d(d),
                                nn.ReLU(inplace=True))
        self.fcs = nn.ModuleList([])
        for i in range(blocks):
            self.fcs.append(
                nn.Conv2d(d, planes * block.expansion, kernel_size=1, stride=1)
            )
        self.softmax = nn.Softmax(dim=1)_**

    def forward(self, x):
        for i, block in enumerate(self.layers):
            residual, out, x = block(x)  #[b,c,w,h]
            **_fea_Zi = out.clone()
            if i == 0:
              fea_Z = fea_Zi.unsqueeze_(dim=1)
            else:
              fea_Z = torch.cat([fea_Z, fea_Zi.unsqueeze_(dim=1)], dim=1)

        fea_F = torch.sum(fea_Z, dim=1)
        fea_u = self.gap(fea_F)
        fea_v = self.fc(fea_u)

        for i, fc in enumerate(self.fcs):
            vector = fc(fea_v)
            if i == 0:
                vectors = vector.unsqueeze_(dim=1)
            else:
                vectors = torch.cat([vectors, vector.unsqueeze_(dim=1)], dim=1)
        vectors = self.softmax(vectors)
        fea_O = F.relu(torch.sum(fea_Z * vectors, dim=1))_**

        return fea_O

**Comment by yan-ctrl** [comment]
Ok, I see. Thank you very much for your timely reply
